base/inference.jl

```diff
@@ -4109,6 +4163,7 @@ function inlining_pass(e::Expr, sv::InferenceState)
                         res = inlining_pass(e, sv)
                     else
                         e.args = Any[GlobalRef(Main.Base,:*), Expr(:call, GlobalRef(Main.Base,:*), a1, a1), a1]
+                        e.args[2].typ = e.typ
                         res = inlining_pass(e, sv)
```

Comment:
That's good to know. There's a pass in inlining that tries to ensure the type information is synchronized, so it's working as intended. But it's just slightly more robust to ensure the information is correct here too.